diff --git a/dist/langgraph.js b/dist/langgraph.js
index cecdea91e40e0a002263dca450a9221d83369498..006a3003be20b02066980c9bcfa2f76fda339c91 100644
--- a/dist/langgraph.js
+++ b/dist/langgraph.js
@@ -129,6 +129,9 @@ var LangGraphAgent = class extends langgraph.LangGraphAgent {
     }
     if ("copilotkit:emit-messages" in (rawEvent.metadata || {})) {
       if (rawEvent.metadata["copilotkit:emit-messages"] === false && isMessageEvent) {
+        if (this.activeRun?.id) {
+          this.messagesInProcess[this.activeRun.id] = null;
+        }
         return false;
       }
     }
diff --git a/dist/langgraph.mjs b/dist/langgraph.mjs
index 4bde33ebf2c8d60a59528f6593c6445da8cdd7ef..ab048f6a7ef8700c7dc2983c3aa60f82158f139b 100644
--- a/dist/langgraph.mjs
+++ b/dist/langgraph.mjs
@@ -128,6 +128,9 @@ var LangGraphAgent = class extends LangGraphAgent$1 {
     }
     if ("copilotkit:emit-messages" in (rawEvent.metadata || {})) {
       if (rawEvent.metadata["copilotkit:emit-messages"] === false && isMessageEvent) {
+        if (this.activeRun?.id) {
+          this.messagesInProcess[this.activeRun.id] = null;
+        }
         return false;
       }
     }
